CHANGELOG.md

```diff
@@ -2,6 +2,21 @@
 
 * enhancements
   * Add support for Ruby 3.1/3.2.
+  * Add support for Hotwire + Turbo, default in Rails 7+.
+    * `:turbo_stream` is now treated as a navigational format, so it works like HTML navigation when using Turbo. Note: if you relied on `:turbo_stream` to be treated as a non-navigational format before, you can reconfigure your `navigational_formats` in the Devise initializer file to exclude it.
+    * Devise requires the latest `responders` version, which allows configuring the status used for validation error responses (`error_status`) and for redirects after POST/PUT/PATCH/DELETE requests (`redirect_status`). For backwards compatibility, Devise keeps `error_status` as `:ok` which returns a `200 OK` response, and `redirect_status` to `:found` which returns a `302 Found` response, but you can configure it to return `422 Unprocessable Entity` and `303 See Other` to match the behavior expected by Hotwire/Turbo:
+
+     ```ruby
+     # config/initializers/devise.rb
+     Devise.setup do |config|
+       # ...
+       config.responder.error_status = :unprocessable_entity
+       config.responder.redirect_status = :see_other
```

Comment:
What are your thoughts on setting these automatically `if defined?(Turbo)`, that way it's automatic for new users and they don't have to modify the config at all?

Reply:
[New users should get the devise initializer setup like that](https://github.com/heartcombo/devise/pull/5548/files#diff-1a65f2d7525a89107811cc734b24f8c92d7a7e04d0265b49c20f97fd928c4f68R299-R306), so they wouldn't have to do any extra step other than running the install generator; however existing users/apps will need to opt-in via the config, so upgrading Devise won't cause any behavior change until they add that config explicitly.

Were you thinking about that use-case? I guess we could do it based on some `defined?(Turbo)` check of sorts, but it _could_ change some unwanted behavior for someone upgrading in that case (and I still want to make sure new users always get the new behavior, whether or not they have Turbo) -- trying to think it through here if there's any scenario, for example they have turbo in their app but have explicitly disabled turbo in Devise forms (because it doesn't work yet), now upgrading Devise could change the responses behavior (because `defined?(Turbo)` is true) and possibly break some client expectation?

Reply:
Ah! I missed the initializer change. I think this is the safer approach.

Some existing apps might use a SPA frontend and expect certain response codes, so changing that would be bad for existing apps.

Reply:
Yeah I agree this approach is sensible.

Given this is a fairly big change and it's not automatic (so people targeting https://github.com/heartcombo/devise/pull/5340 will need to make this change when they update), what do you think adding a `post_install_message` to the Gemspec to draw people's attention to it?

Reply:
> what do you think adding a post_install_message to the Gemspec to draw people's attention to it?

I think we can add a temporary message there for this new release. I haven't been much a fan of those messages popping up on every gem install, but this one might be worth an exception.

Reply:
Ok, I just pushed a post install message, and added a [wiki](https://github.com/heartcombo/devise/wiki/How-To:-Upgrade-to-Devise-4.9.0-[Hotwire-Turbo-integration]) with some additional information for upgrading (basically expanding what's already on the changelog, but it makes it easier to update / tweak if needed). It's a little annoying because I can't enforce the responders version just yet, but hopefully that's clear enough.

Let me know if you can think of anything else, otherwise I think we're mostly done. Thanks!